examples/particles.rs

```diff
@@ -0,0 +1,309 @@
+
+        server.clients.retain(|_, client| {
+            if client.created_this_tick() {
+                if self
+                    .player_count
+                    .fetch_update(Ordering::SeqCst, Ordering::SeqCst, |count| {
+                        (count < MAX_PLAYERS).then_some(count + 1)
+                    })
+                    .is_err()
+                {
+                    client.disconnect("The server is full!".color(Color::RED));
+                    return false;
+                }
+
+                match server
+                    .entities
+                    .insert_with_uuid(EntityKind::Player, client.uuid(), ())
+                {
+                    Some((id, _)) => client.state = id,
+                    None => {
+                        client.disconnect("Conflicting UUID");
+                        return false;
+                    }
+                }
+
+                client.spawn(world_id);
+                client.set_flat(true);
+                client.set_game_mode(GameMode::Creative);
+                client.teleport(
+                    [
+                        SPAWN_POS.x as f64 + 0.5,
+                        SPAWN_POS.y as f64 + 1.0,
+                        SPAWN_POS.z as f64 + 0.5,
+                    ],
+                    0.0,
+                    0.0,
+                );
+                client.set_player_list(server.state.player_list.clone());
+
+                if let Some(id) = &server.state.player_list {
+                    server.player_lists.get_mut(id).insert(
+                        client.uuid(),
+                        client.username(),
+                        client.textures().cloned(),
+                        client.game_mode(),
+                        0,
+                        None,
+                    );
+                }
+            }
+
+            if client.is_disconnected() {
+                self.player_count.fetch_sub(1, Ordering::SeqCst);
+                if let Some(id) = &server.state.player_list {
+                    server.player_lists.get_mut(id).remove(client.uuid());
+                }
+                server.entities.remove(client.state);
+
+                return false;
+            }
+
+            let entity = server
+                .entities
+                .get_mut(client.state)
+                .expect("missing player entity");
+
+            while handle_event_default(client, entity).is_some() {}
+
+            true
+        });
+
+        if !server.clients.is_empty() && server.shared.current_tick() % 30 == 0 {
+            if server.state.particle_index == server.state.particle_list.len() {
+                server.state.particle_index = 0;
+            }
+            let pos = Vec3::new(0.0, 100.0, -10.0);
+            let offset = Vec3::new(0.5, 0.5, 0.5);
+            let opt: Option<&ParticleType> =
+                server.state.particle_list.get(server.state.particle_index);
```

Comment:
Since you're panicking in the `None` case anyway, it may be more idiomatic to use `.expect()` or just index the vec. Up to you.